src/framework/handlers/esmscript.js

```diff
@@ -0,0 +1,103 @@
+import { AssetRegistry } from '../asset/asset-registry.js';
+
+export class ScriptCache {
+    static _scripts = new Map();
+
+    static register(path, script) {
+        const url = URL.canParse(path) ? new URL(path) : new URL(path, 'file://');
+        ScriptCache._scripts.set(url.pathname, script);
+    }
+
+    static get(path) {
+        const url = URL.canParse(path) ? new URL(path) : new URL(path, 'file://');
+        return ScriptCache._scripts.get(url.pathname);
+    }
+
+    static clear() {
+        ScriptCache._scripts.clear();
+    }
+}
+
+/**
+ * This custom import statement for ES Modules conditionally
+ * checks and resolves modules in bundled or non bundled modules
+ *
+ * **The api is likely to change, use at your own discretion**
+ *
+ * @param {import('../app-base').AppBase} app - The application scope to load from.
+ * @param {string} moduleSpecifier - A unique path or specifier used to import the module
+ * @returns {Promise} Returns a promise which fulfills to a module namespace object: an object containing all exports from moduleName.
+ *
+ * @todo add support for bundle contexts
+ * @ignore
+ */
+export const DynamicImport = (app, moduleSpecifier) => {
+
+    // If the `AssetRegistry.assetBaseUrl` is defined, use it as the base URL for the import.
+    const baseUrl = AssetRegistry?.assetBaseUrl ?? import.meta.url;
+    const isFileProtocol = import.meta.resolve(moduleSpecifier).startsWith('file:');
+
+    // If we are loading a local file, we don't need to append the baseUrl.
+    const path = isFileProtocol ?
+        moduleSpecifier :
+        `${baseUrl}/${moduleSpecifier}`;
```

Comment:
@slimbuck This is now using a runtime path check, where `AssetRegistry.assetBaseUrl` should specify the full base URL. I've kept this separate to `AssetRegistry.prefix` which is only a relative path. We'll need something in the launcher that sets this conditionally if `use_local_engine` is present, but this is not necessary for prod